### PR TITLE
Welcome Dialog: Add ability to set Dark Theme

### PR DESCRIPTION
**Pull request by elad335**
Add the ability to set dark theme in the welcome dialog, set dark by default if the system-wide theme is already dark.
Probably the most important PR in RPCS3 history.

image

### CODE REVIEW

**Comment by elad335** [comment]
Seems to work under Windows 10, idk how about Linux and MacOS. (the ddetection of darkness of the default system theme)

**Comment by Megamouse** [comment]
I think this is all overcomplicated...
I believe you can just check:
`if (QGuiApplication::styleHints()->colorScheme() == Qt::ColorScheme::Dark)`

File: rpcs3/rpcs3qt/gui_application.cpp [hunk 1]
```diff
@@ -135,10 +135,19 @@ bool gui_application::Init()
 	// Create connects to propagate events throughout Gui.
 	InitializeConnects();
 
+	bool set_dark_theme = false;
+
 	if (m_gui_settings->GetValue(gui::ib_show_welcome).toBool())
 	{
 		welcome_dialog* welcome = new welcome_dialog(m_gui_settings, false);
```
Comment (pedroalexandr) [inline rpcs3/rpcs3qt/gui_application.cpp:142]: You have to delete this pointer before it goes out of scope, this block of code is memory leaking.

Comment (elad335) [reply rpcs3/rpcs3qt/gui_application.cpp:142]: It's not, QT takes care of it.

Comment (pedroalexandr) [reply rpcs3/rpcs3qt/gui_application.cpp:142]: Woah, didn't know that. Sorry.

File: rpcs3/Emu/perf_monitor.cpp [hunk 2]
```diff
@@ -14,20 +14,44 @@ void perf_monitor::operator()()
...
+		const bool is_paused = Emu.IsPaused();
 
+		if (!is_paused)
+		{
+			logged_pause = 0;
+		}
+
+		elapsed_us += update_interval_us;
+
+		per_core_usage.clear();
```
Comment (Megamouse) [inline rpcs3/Emu/perf_monitor.cpp:34]: this makes no sense, as the vector is resized in the next function

File: rpcs3/Emu/perf_monitor.cpp [hunk 3]
```diff
@@ -14,20 +14,44 @@ void perf_monitor::operator()()
...
-		elapsed_us += update_interval_us;
 
-		double total_usage = 0.0;
-		std::vector<double> per_core_usage;
+		const bool is_paused = Emu.IsPaused();
 
+		if (!is_paused)
+		{
+			logged_pause = 0;
+		}
```
Comment (Megamouse) [inline rpcs3/Emu/perf_monitor.cpp:30]: this entire section is misplaced here and needs to go in the scope below.

File: rpcs3/Emu/perf_monitor.cpp [hunk 4]
```diff
@@ -14,20 +14,44 @@ void perf_monitor::operator()()
...
 	stats.init_cpu_query();
 
+	u32 logged_pause = 0;
+
+	std::vector<double> per_core_usage;
+
 	while (thread_ctrl::state() != thread_state::aborting)
 	{
 		thread_ctrl::wait_for(update_interval_us);
-		elapsed_us += update_interval_us;
```
Comment (Megamouse) [inline rpcs3/Emu/perf_monitor.cpp:20]: I don't see any reason that this line was changed

File: rpcs3/util/cpu_stats.cpp [hunk 5]
```diff
@@ -126,8 +126,8 @@ namespace utils
 	{
 		total_usage = 0.0;
 
+		per_core_usage.clear();
```
Comment (Megamouse) [inline rpcs3/util/cpu_stats.cpp:129]: What are these random changes?
resize + fill is faster than clear + resize, according to quickbench

File: rpcs3/util/dyn_lib.hpp [hunk 6]
```diff
@@ -60,14 +60,20 @@ namespace utils
 	// (assume the lib is always loaded)
 	void* get_proc_address(const char* lib, const char* name);
 
-	template <typename F>
+	template <typename CallingConventionTag, typename F>
```
Comment (Megamouse) [inline rpcs3/util/dyn_lib.hpp:63]: All the changes in this file are neither explained nor have any relation to the rest of the code

**Comment by Megamouse** [comment]
Also, please stop shoving multiple random changes into the same PR.
This is getting really frustrating.

**Pull request merged**